exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/ProfileResources.java

```diff
@@ -220,7 +225,21 @@ public QProfiles getProfilesJSON() {
 
       final List<ProfileInfo> finishedQueries = Lists.newArrayList();
 
-      final Iterator<Map.Entry<String, QueryProfile>> range = completed.getRange(0, MAX_PROFILES);
+      //Defining #Profiles to load
+      int maxProfilesToLoad;
+      if (MaxProfiles == null) {
+        MaxProfiles = work.getContext().getConfig().getInt(ExecConstants.HTTP_MAX_PROFILES);
+      }
+
+      String maxProfilesParams = uriInfo.getQueryParameters().getFirst(MAX_QPROFILES_PARAM);
```

Comment:
You can use annotations for the property. That's the best way in this particular context.